codespell_lib/tests/test_dictionary.py

```diff
@@ -1,61 +1,182 @@
+                                   'with trailing ","' % (err,))
+    reps = [r.strip() for r in rep.lower().split(',')]
+    reps = [r for r in reps if len(r)]
+    for r in reps:
+        assert err != r.lower(), ('error %r corrects to itself amongst others'
+                                  % (err,))
+        _check_aspell(
+            r, 'error %s: correction %r' % (err, r), in_aspell[1], fname)
+    assert len(set(reps)) == len(reps), 'entries are not (lower-case) unique'
+
+
+@pytest.mark.parametrize('err, rep, match', [
+    ('a a', 'bar', 'has whitespace'),
+    ('a,a', 'bar', 'has a comma'),
+    ('a', '', 'non-empty'),
+    ('a', ' bar', 'start with whitespace'),
+    ('a', ',bar', 'starts with a comma'),
+    ('a', 'bar,bat', '.*not.*followed by a space'),
+    ('a', 'bar ', 'trailing space'),
+    ('a', 'b ,ar', 'contains a whitespace.*followed by a comma'),
+    ('a', 'bar,', 'single entry.*comma'),
+    ('a', 'bar, bat', 'must end with trailing ","'),
+    ('a', 'a, bar,', 'corrects to itself amongst others'),
+    ('a', 'a', 'corrects to itself'),
+    ('a', 'bar, bar,', 'unique'),
+])
+def test_error_checking(err, rep, match):
+    """Test that our error checking works."""
+    with pytest.raises(AssertionError, match=match):
+        _check_err_rep(err, rep, (None, None), 'dummy')
+
+
+@pytest.mark.parametrize('err, rep, err_aspell, rep_aspell, match', [
+# This doesn't raise any exceptions, so skip for now
+#    pytest.param('a', 'uvw, bar,', None, None, 'should be in aspell', marks=[
+#        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('abc', 'uvw, bar,', True, None, 'should be in aspell', marks=[
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('a', 'uvw, bar,', False, None, 'should not be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('a', 'abc, uvw,', None, True, 'should be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('abc', 'uvw, bar,', True, True, 'should be in aspell', marks=[
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('abc', 'uvw, bar,', False, True, 'should be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('a', 'bar, back,', None, False, 'should not be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('abc', 'uvw, xyz,', True, False, 'should be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
+    pytest.param('abc', 'uvw, bar,', False, False, 'should not be in aspell', marks=[  # noqa: E501
+        pytest.mark.skipif(speller is None, reason='requires aspell')]),
```

Comment:
If this whole test requires `aspell` then you don't need to mark all individually you can just mark the whole thing, I'll push a commit